### PR TITLE
Portal resource unit updates / Harvest waste type

### PR DESCRIPTION
**Pull request by dandehavilland**
- Updated stubbed data in plant harvest and package start to use latest from portal
- Updated resource unit lookups to use metrc-specific portal updates
- Added rake task to sync 
- Use `product_modifier` from waste resource unit if it is allowed by Metrc
  - Validate supplied waste type against Metrc supported waste types, raise exception (=> notify CS) if not found, suggest close matches if possible.

---

TODO
- [x] Document rake script

### CODE REVIEW

File: app/services/metrc_service/plant/harvest.rb [hunk 1]
```diff
@@ -74,10 +74,21 @@ def build_remove_waste_payload
...
 
-      def waste_type(_completion)
-        # TODO: determine waste type from 'process' completion
-
-        'Plant Material'
+      # If the waste type (the resource unit's +product_modifier+) is found in the list of available waste types
+      # from Metrc (GET [metrc]/harvests/v1/waste/types), # then we will use that.
+      #
+      # If not, then we use the first available valid waste type from Metrc
+      def waste_type(completion)
```
Comment (dandehavilland) [inline app/services/metrc_service/plant/harvest.rb:81]: @sd @betacar could you please check this method and let me know if this aligns with your thinking

File: spec/services/metrc_service/base_spec.rb [hunk 2]
```diff
@@ -258,43 +258,49 @@
...
+    include_context 'with synced data' do
+      let(:facility_id) { 1 }
+      let(:batch_id) { 331 }
+    end
+
     before do
-      stub_request(:get, 'https://portal.artemisag.com/api/v3/facilities/1568')
-        .to_return(body: { data: { id: '1568', type: 'facilities', attributes: { id: 1568, name: 'Rare Dankness' } } }.to_json)
+      stub_request(:get, "https://portal.artemisag.com/api/v3/facilities/#{facility_id}")
+        .to_return(body: load_response_json("api/sync/facilities/#{facility_id}"))
```
Comment (betacar) [inline spec/services/metrc_service/base_spec.rb:268]: Can you please move the `load_response_json`. and maybe rename it to `load_stubbed_json`, to a spec helper method?

File: spec/services/metrc_service/batch_spec.rb [hunk 3]
```diff
@@ -50,10 +50,8 @@
...
... { id: 6666, slug: 'flower-room-barcoded', name: 'Flower Room Barcoded', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6422, slug: 'mothers', name: 'Mothers', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6434, slug: 'mothers', name: 'Mothers', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6427, slug: 'propagation', name: 'Propagation', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6424, slug: 'vegetation', name: 'Vegetation', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6430, slug: 'vegetation-field', name: 'Vegetation Field', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6428, slug: 'vegetation-greenhouse', name: 'Vegetation Greenhouse', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }, { id: 6665, slug: 'veg-room-barcoded', name: 'Veg Room Barcoded', seeding_unit_id: 3479, zone_type: 'generic', sub_zones: [] }], secondary_display_active: nil, secondary_display_capacity: nil }, seeding_unit_capacity: 0, system: 'None' }, relationships: { sub_zones: { meta: { included: false } }, seeding_unit: { meta: { included: false } } } }] }.to_json)
```
Comment (betacar) [inline spec/services/metrc_service/batch_spec.rb:51]: Let's move this stubbed data to a separate JSON file, please.

File: spec/services/metrc_service/batch_spec.rb [hunk 4]
```diff
@@ -69,6 +67,9 @@
         stub_request(:get, 'https://portal.artemisag.com/api/v3/facilities/1568/batches/2002?include=zone,barcodes,completions,custom_data,seeding_unit,harvest_unit,sub_zone')
```
Comment (betacar) [inline spec/services/metrc_service/batch_spec.rb:67]: Same as above, please.

File: spec/services/metrc_service/base_spec.rb (hunk 2 above)
Comment (betacar) [reply spec/services/metrc_service/base_spec.rb:268]: nvm

**Comment by dandehavilland** [comment]
@betacar I've reworked the `waste_type` to provide some validation and information to CS when incorrect. Please re-review when you have a moment.

**Pull request merged**